glslc/src/file_compiler.cc

```diff
@@ -177,22 +170,28 @@ void FileCompiler::OutputMessages() {
   shaderc_util::OutputMessages(&std::cerr, total_warnings_, total_errors_);
 }
 
+namespace {
+// A helper function to return the base file name from either absolute path or
+// relative path representation of a file.
+// e.g.: dir_a/dir_b/file_c.vert => file_c.vert
+std::string GetBaseFileName(const std::string& file_path) {
+  return file_path.substr(file_path.find_last_of("/\\") + 1);
```

Comment:
Move this to `libshaderc_util::io` and test properly.  This version doesn't work for `file_path` with no backslashes or ending in backslash.